src/backend/linux_raw/fs/raw_dir.rs

```diff
@@ -0,0 +1,204 @@
+use core::cmp::max;
+use core::convert::TryFrom;
+use core::mem::MaybeUninit;
+use core::slice;
+use linux_raw_sys::general::linux_dirent64;
+
+use crate::backend::fs::syscalls::getdents_uninit;
+use crate::fd::AsFd;
+use crate::ffi::CStr;
+use crate::fs::FileType;
+use crate::io;
+
+/// Suggested buffer size for use in [`RawDir::new`]. There is no requirement to
+/// use this buffer size, but it was found to be a good tradeoff between memory
+/// use and syscall count.
+pub const DIR_BUF_LEN: usize = 8192;
+
+/// A directory iterator implemented with getdents.
+///
+/// This implementation:
+/// - Excludes deleted inodes (with ID 0).
```

Comment:
Could you say more about the motivation for excluding entries with inode 0? I don't see any mention of this in the `getdents` documentation. 

From some searching, it looks like glibc historically skipped entries with d_ino == 0, but has recently been changed to not skipping them:
 - https://sourceware.org/bugzilla/show_bug.cgi?id=12165
 - https://sourceware.org/pipermail/libc-alpha/2022-September/142059.html

Reply:
Oh wow, that discussion is super recent. Thanks for looking into this! My implementation was derived from [glibc's readdir64_r](https://github.com/bminor/glibc/blob/b92a49359f33a461db080a33940d73f47c756126/sysdeps/unix/sysv/linux/readdir64_r.c) which hasn't merged in that patch I guess. Either way, happy to remove the inode check.